staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -68,29 +69,39 @@ const (
 )
 
 // NewLeaderElector creates a LeaderElector from a LeaderElectionConfig
-func NewLeaderElector(lec LeaderElectionConfig) (*LeaderElector, error) {
+func NewLeaderElector(lec LeaderElectionConfig) (*LeaderElector, []error) {
```

Comment:
I see little value in this patch and you would need to update all the call sites.  
The only way you could/would have cascading errors is on a misconfiguration or failed ntp deployments. 

/hold